docs/user_docs/guides/ci-integration/2-gitlab-ci.md

```diff
@@ -0,0 +1,135 @@
+    - git commit -am '[skip ci] image update to $CI_REGISTRY_IMAGE:$CI_COMMIT_SHORT_SHA'
+    - git push origin main
+  only:
+    - main
+```
+
+We need the workflow in the source code repository to automatically trigger the workflow in the deployment manifest repository. At this point, we need to config `Settings -> CI/CD -> Variables` including `CI_REGISTRY`, `CI_REGISTRY_IMAGE`, `CI_REGISTRY_USER`, `CI_REGISTRY_PASSWORD`, `CI_USERNAME` and `CI_PASSWORD` and update application code to trigger automation build and deploy.
+
+![](/img/docs/user_docs/guides/ci-integration/gitlab-ci-variables.jpg)
+
+### 2. Commit the Application Code
+
+After submitting in the `flask-demo` repository, Gitlab will automatically build a container image and push it to the Docker hub. It will also then trigger the Action of the `flask-demo-kcl-manifest` repository and modify the image value in the deployment manifest repository through [KCL Automation API](/docs/user_docs/guides/automation). Now let's create a submission in the `flask-demo` repository, and we can see that the code submission triggers the Gitlab CI process for the application repository `Build -> Pipelines` page.
+
+### 3. Configuration Automatic Update
+
+After the Gitlab CI process in the application repository is completed, a automatic update configuration CI will be triggered in the repository where the KCL configuration is stored and submitted to the main branch of the `flask-demo` repository.
+
++ We can obtain the deployment manifest source code for compilation and validation
+
+```shell
+git checkout main && git pull && cd deploy && kcl
+```
+
+The output YAML is
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: flask_demo
+  labels:
+    app: flask_demo
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: flask_demo
+  template:
+    metadata:
+      labels:
+        app: flask_demo
+    spec:
+      containers:
+        - name: flask_demo
+          image: "kcllang/flask_demo:6428cff4309afc8c1c40ad180bb9cfd82546be3e"
+          ports:
+            - protocol: TCP
+              containerPort: 5000
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: flask_demo
+  labels:
+    app: flask_demo
+spec:
+  type: NodePort
+  selector:
+    app: flask_demo
+  ports:
+    - port: 5000
+      protocol: TCP
+      targetPort: 5000
+```
+
+From the above configuration, it can be seen that the image of the resource is indeed automatically updated to the newly constructed image value. In addition, we can also use the **Argo CD KCL plugin** to automatically synchronize data from the Git repository and deploy the application to the Kubernetes cluster.
+
+## Summary
+
+By integrating KCL and Gitlab CI, we can automate the modification and deployment configuration of any application output container image, in order to achieve end-to-end application development process and improve R&D deployment efficiency.
```

Comment:
> we can automate the modification and deployment configuration of any application output container image

Maybe it's better:
we can integrate the container build and delivery workflow by automatically updating the image values in the configuration

Reply:
> > we can automate the modification and deployment configuration of any application output container image
> 
> Maybe it's better: we can integrate the container build and delivery workflow by automatically updating the image values in the configuration

fixed.